logstash-core/src/main/java/org/logstash/config/ir/PipelineIR.java

```diff
@@ -51,7 +51,7 @@ public PipelineIR(Graph inputSection, Graph filterSection, Graph outputSection,
 
         this.graph.validate();
 
-        if (this.getOriginalSource() != null && this.getOriginalSource().matches("^\\S+$")) {
+        if (this.getOriginalSource() != null && !this.getOriginalSource().matches("^\\s+$")) {
```

Comment:
I'm not famililar with this code but doesn't this regexp test if `this.getOriginalSource()` source is made up _entirely_ of whitespace? Should it be something like `.matches("\\s")` instead to check if there is _any_ whitespace in `this.getOriginalSource()` ?